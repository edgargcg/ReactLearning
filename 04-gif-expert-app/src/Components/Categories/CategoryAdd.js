import React, {  useState } from 'react'
import PropTypes from 'prop-types';

export const CategoryAdd = ({ setCategory }) => {
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        const { value } = e.target;
        setInputValue(value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2){
            setCategory(inputValue);
            setInputValue('');
        }
    }

    return (    
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"  
                value={inputValue}
                placeholder="Category"
                onChange={ handleInputChange }
            />
        </form>
    )
}

CategoryAdd.propTypes = {
    setCategory: PropTypes.func.isRequired
}