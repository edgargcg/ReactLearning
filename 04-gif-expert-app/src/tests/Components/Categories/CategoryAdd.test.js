import { shallow } from 'enzyme';
import { CategoryAdd } from "../../../Components/Categories/CategoryAdd";

describe('Test on <CategoryAdd />', () => {
    const setCategory = jest.fn();
    let wrapper = shallow(<CategoryAdd setCategory={setCategory} />);

    beforeEach(
        () => {
            jest.clearAllMocks();
            wrapper = shallow(<CategoryAdd setCategory={setCategory} />);
        }
    );

    test('should render <CategoryAdd />', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have a value on type on input', () => {
        const value = 'batman';
        const input = wrapper.find('input');
        
        input.simulate('change', { target: { value } });

        const inputAfterChange = wrapper.find('input');
        const result = inputAfterChange.props().value;
        
        expect(result).toBe(value);
    })
    
    test('should not get the data on submit', () => {
        wrapper.find('form').simulate(
            'submit', 
            { preventDefault(){} }
        );
        
        expect(setCategory).not.toHaveBeenCalled();
    })
    
    test('should call setCategory and clean input control', () => {
        const value = 'batman';
        wrapper
            .find('input')
            .simulate('change', { target: { value } });

        wrapper
            .find('form')
            .simulate('submit', { preventDefault(){} });
        
        expect(setCategory).toHaveBeenCalled();

        const finalValue = wrapper.find('input').props().value;

        expect(finalValue).toBe('');    
    })
    

})
