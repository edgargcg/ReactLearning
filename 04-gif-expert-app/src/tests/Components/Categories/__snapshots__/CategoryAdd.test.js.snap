// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <CategoryAdd /> should render <CategoryAdd /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Category"
    type="text"
    value=""
  />
</form>
`;
